doc/contributing.rst

```diff
@@ -0,0 +1,874 @@
+
+#. Explain why the current behavior is wrong/not desired and what you expect instead.
+
+The issue will then show up to the *xarray* community and be open to comments/ideas from others.
+
+.. _contributing.github:
+
+Working with the code
+=====================
+
+Now that you have an issue you want to fix, enhancement to add, or documentation to improve,
+you need to learn how to work with GitHub and the *xarray* code base.
+
+.. _contributing.version_control:
+
+Version control, Git, and GitHub
+--------------------------------
+
+To the new user, working with Git is one of the more daunting aspects of contributing to *xarray*.
+It can very quickly become overwhelming, but sticking to the guidelines below will help keep the process
+straightforward and mostly trouble free.  As always, if you are having difficulties please
+feel free to ask for help.
+
+The code is hosted on `GitHub <https://www.github.com/pydata/xarray>`_. To
+contribute you will need to sign up for a `free GitHub account
+<https://github.com/signup/free>`_. We use `Git <http://git-scm.com/>`_ for
+version control to allow many people to work together on the project.
+
+Some great resources for learning Git:
+
+* the `GitHub help pages <http://help.github.com/>`_.
+* the `NumPy's documentation <http://docs.scipy.org/doc/numpy/dev/index.html>`_.
+* Matthew Brett's `Pydagogue <http://matthew-brett.github.com/pydagogue/>`_.
+
+Getting started with Git
+------------------------
+
+`GitHub has instructions <http://help.github.com/set-up-git-redirect>`__ for installing git,
+setting up your SSH key, and configuring git.  All these steps need to be completed before
+you can work seamlessly between your local repository and GitHub.
+
+.. _contributing.forking:
+
+Forking
+-------
+
+You will need your own fork to work on the code. Go to the `xarray project
+page <https://github.com/pydata/xarray>`_ and hit the ``Fork`` button. You will
+want to clone your fork to your machine::
+
+    git clone https://github.com/your-user-name/xarray.git xarray-yourname
+    cd xarray-yourname
+    git remote add upstream https://github.com/pydata/xarray.git
+
+This creates the directory `xarray-yourname` and connects your repository to
+the upstream (main project) *xarray* repository.
+
+.. _contributing.dev_env:
+
+Creating a development environment
+----------------------------------
+
+To test out code changes, you'll need to build xarray from source, which
+requires a Python environment. If you're making documentation changes, you can
+skip to :ref:`contributing.documentation` but you won't be able to build the
+documentation locally before pushing your changes.
+
+.. _contributiong.dev_python:
+
+Creating a Python Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before starting any development, you'll need to create an isolated xarray
+development environment:
+
+- Install either `Anaconda <https://www.anaconda.com/download/>`_ or `miniconda
+  <https://conda.io/miniconda.html>`_
```

Comment:
Is it worth mentioning that `pip` is an acceptable alternative?

Reply:
I think we can add that later if/when there is interest. The requirements files in ci/ are formatted for conda so I suggest we leave this as is for now.

Reply:
👍 

Tangentially, are we the only ppl who don't use conda?

Reply:
Perhaps. Personally, conda has become pretty baked into my workflow but I know that is not the case for everyone. 

If you want to push a description of how you setup your development environment, I think that would be a welcome addition here.

Reply:
Personally I lost faith in conda some time ago when the [sky fell](https://github.com/conda-forge/pyproj-feedstock/issues/10). Since wheels are available for most packages, installing with pip is (almost) as easy as with conda nowadays.

Reply:
@jhamman interesting. TBC, I didn't mean that I had any edge here. Fewer options are better - I vote to leave conda only.

Any hardcore pippers can deal with the issues themselves